File: backend/VocabuAI.Api/Dtos/FlashCards/FlashCardEditDto.cs
namespace VocabuAI.Api.Dtos.FlashCards;

// Keep in sync with frontend/src/domain/dtos/flashcards/FlashCardEditDto.ts

public sealed record FlashCardEditDto(
    int Id,
    string ForeignLanguage,
    string LocalLanguage,
    string ForeignLanguageCode,
    string LocalLanguageCode,
    string? Synonyms,
    string? Annotation
);

File: frontend/src/domain/dtos/flashcards/FlashCardEditDto.ts
// Keep in sync with backend/VocabuAI.Api/Dtos/FlashCards/FlashCardEditDto.cs
export type FlashCardEditDto = {
  id: number;
  foreignLanguage: string;
  localLanguage: string;
  foreignLanguageCode: string;
  localLanguageCode: string;
  synonyms?: string | null;
  annotation?: string | null;
};
